# The Solar Energy Breakthrough

### The Scenario:
The Federal Network Agency states that the standard residential solar pandels produce a national historical average of 2.50 kWh of energy per day during early autumn.

A University has partnered with "The Engineering Department" to install a new,experimental light-capturing films over the solar panels on the university roof and they want to mathematically prove this new film is breakthrough.

### Objective:
Prove mathematically that The University's solar energy production average is more than the national average.

>Test Used : Z-test

>Test performed by : Raghav Katyal as on 4 October 2026

In [66]:
import numpy as np
from scipy import stats
from statsmodels.stats.weightstats import ztest

In [67]:
# University Solar Energy Data of 40 days
np.random.seed(42)
data = np.random.normal(loc=2.8,scale=0.4,size=40)
display(data)

array([2.99868566, 2.74469428, 3.05907542, 3.40921194, 2.70633865,
       2.70634522, 3.43168513, 3.10697389, 2.61221025, 3.01702402,
       2.61463292, 2.6137081 , 2.89678491, 2.0346879 , 2.11003287,
       2.57508499, 2.39486755, 2.92569893, 2.43679037, 2.23507852,
       3.38625951, 2.70968948, 2.82701128, 2.23010073, 2.58224691,
       2.84436904, 2.33960257, 2.95027921, 2.55974452, 2.6833225 ,
       2.55931736, 3.54091127, 2.79460111, 2.37691563, 3.12901796,
       2.31166254, 2.88354544, 2.01613195, 2.26872558, 2.87874449])

In [68]:
#Running Shapiro Test to verify the data is normal
w_value,p_value = stats.shapiro(data)
print(f"W-Statistic : {w_value:.3f}")
print(f"P-value : {p_value:.4f}")

W-Statistic : 0.979
P-value : 0.6604


>Observation : A W-value of 0.979 and a p-value of 0.6604 shows no strong evidence against Normality of data ; hence test is passed. Data is considered normally distributed.

In [69]:
z_score,p_value = ztest(x1=data,value=2.5)
print(f"Z-score : {z_score:.3f}")
print(f"P-value : {p_value:.4f}")

Z-score : 3.527
P-value : 0.0004


>Observation: A Z-score of 3.527 and p-value of 0.0004 shows evidence against matching the global average ; data fails the test. Thus, it is mathematically observed that : The University's solar average are more than the national average. It is indeed breakthrough.